pwiz_aux/msrc/utility/vendor_api/Waters/WatersRawFile.hpp

```diff
@@ -569,7 +569,15 @@ struct PWIZ_API_DECL RawData
 
         for (int ch = 0; ch < channels; ch++)
         {
-            AnalogChromatogramReader.ReadChannel(ch, analogTimes[ch], analogIntensities[ch]);
+            try
+            {
+                AnalogChromatogramReader.ReadChannel(ch, analogTimes[ch], analogIntensities[ch]);
+            }
+            catch(MassLynxRawException const& ex)
+            {
+                // Likely just an empty channel - if there's a real problem the folllowing calls will probably throw too
+                std::cout << ex.what() << std::endl;

```

Comment:
Use cerr without std:: prefix (#include pwiz/utility/misc/Stream.hpp if needed) and add a TODO(MCC) to switch to BOOST_LOG. Probably add a prefix to the error message like `"Error reading channel " << ch << ": "`

So when this happens, analogTimes[ch] is left empty, or is it partially filled? The goal here is to produce an empty chromatogram in the mzML?

Reply:
Ah, jumped the gun there. What did the cout say in your testing? Anything helpful? If not outputting it, remove the `ex` variable name to avoid the "unused variable" warning.

Reply:
Nah, my bad - didn't mean to leave that in there. It just gave "generic error", not very useful.

Debugger didn't want to show me the contents of the data arrays, I'll try an mzML conversion and see what we get. My impression was that it would be an empty array but I will verify.

Reply:
Yeah, they just come up empty:

`        <chromatogram index="28" id="Sample Temp" defaultArrayLength="0">
          <cvParam cvRef="MS" accession="MS:1000626" name="chromatogram type" value=""/>
          <binaryDataArrayList count="2">
            <binaryDataArray encodedLength="0">
              <cvParam cvRef="MS" accession="MS:1000523" name="64-bit float" value=""/>
              <cvParam cvRef="MS" accession="MS:1000574" name="zlib compression" value=""/>
              <cvParam cvRef="MS" accession="MS:1000595" name="time array" value="" unitCvRef="UO" unitAccession="UO:0000031" unitName="minute"/>
              <binary></binary>
            </binaryDataArray>
            <binaryDataArray encodedLength="0">
              <cvParam cvRef="MS" accession="MS:1000521" name="32-bit float" value=""/>
              <cvParam cvRef="MS" accession="MS:1000574" name="zlib compression" value=""/>
              <cvParam cvRef="MS" accession="MS:1000515" name="intensity array" value="" unitCvRef="UO" unitAccession="UO:0000027" unitName="degree Celsius"/>
              <binary></binary>
            </binaryDataArray>
          </binaryDataArrayList>
        </chromatogram>`

Reply:
@chambm can I have the blessing, then?